Source/NSApplication.m

```diff
@@ -1259,6 +1259,24 @@ - (void) dealloc
  */
 - (void) activateIgnoringOtherApps: (BOOL)flag
 {
+  NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+  NSString       *autolaunch = [defaults objectForKey: @"autolaunch"];
+
+  /* Application was executed with argument '-autolaunch YES'. Set argument 
+   * value to "NO" for future calls to `activateIgnoringOtherApps:`. */
+  NSLog(@"autolaunch = %@", autolaunch);
```

Comment:
You probably did forget to remove this log statement.

This whole check of the user defaults which gets done on each call of this method looks rather slow to me. Shouldn't we read this value once and keep it in a static variable and only check (and set) that from here?

Reply:
I understand your concern but `-activateIgnoringOtherApps:` method called only on application activation. I think such complication is not the case for optimization.

Yes, I forgot to remove temporary debug output. I'll update code right now.

Reply:
I forgot to mention that leaving `autolaunch` argument in defaults domain is intentional. If some application need to know if it was autolaunched at startup (for example in -applicationDidFinishLaunching:), one should check if [defaults objectForKey:@"autolaunch"] is not `nil`.

Reply:
Hm... I don't know how to update patch in PR. @fredkiefer, could you help me?

Reply:
@fredkiefer , PR has added commit. Please, merge if it's ok.